apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -111,6 +117,43 @@ export class GeneratePreviewUsecase {
     }
   }
 
+  /**
+   * Fixes invalid Liquid template variables for preview by replacing them with error messages.
+   *
+   * @example
+   * // Input controlValues:
+   * { "message": "Hello {{invalid.var}}" }
+   *
+   * // Output:
+   * { "message": "Hello [[Invalid Variable: invalid.var]]" }
+   */
+  private fixInvalidLiquidOutputsForPreview(controlValues: Record<string, unknown>): Record<string, unknown> {
+    const res = extractLiquidTemplateVariables(JSON.stringify(controlValues));
+
+    const { invalidVariables } = res;
+    let controlValuesString = JSON.stringify(controlValues);
+
+    for (const invalidVariable of invalidVariables) {
+      const invalidVariableExists = controlValuesString.includes(invalidVariable.name);
+      if (!invalidVariableExists) {
+        continue;
+      }
+
+      /*
+       * we need to remove the '{{' and '}}' from the invalid variable name
+       * so that framework can parse it without errors
+       */
+      const invalidVariableExpression = invalidVariable.name.replace('{{', '').replace('}}', '');
```

Comment:
self note: need to replace all (the same) invalid outputs at once here.
```suggestion
      const invalidVariableExpression = invalidVariable.name.replaceAll('{{', '').replace('}}', '');
```